src/providers/github.py

```diff
@@ -0,0 +1,40 @@
+from re import I
+
+from github import Github
+import requests
+
+from scanner import scan_diff
+from rules import read_rules
+
+def process_repo( token, repo ):
+    # login to github with our token
+    g = Github(token)
+
+    if repo == None:
+        # We're going to grab the first repo for this user.
+        print("No repo provided, so I'll choose a repo from this user")
+        repo = g.get_user().get_repos()[0]
```

Comment:
This of course copies my original code...  We will have to remove it and exit on error in future code update.